scrapy_poet/downloadermiddlewares.py

```diff
@@ -107,6 +107,13 @@ def process_response(
         and an injectable attribute,
         the user-defined ``cb_kwargs`` takes precedence.
         """
+        if request.callback is None:
+            spider.logger.warning(
+                f"{request} has callback=None. No dependencies will be built "
+                f"by scrapy-poet."
+            )
```

Comment:
Instead of warning here, could we move this to `process_request`, and only warn the first time if the `parse` method has something that would cause action from the injection middleware? (i.e. DummyResponse or something that would be subject to be resolved in process_response)

Reply:
I've updated this part to only warn once when the `parse()` method has been annotated with `DummyResponse`.

I think we should keep the warning here in `process_response()` since there could be custom middlewares in some project that tries to replace the `callback=None` with something else.